### PR TITLE
feat(ui): add name to dashboard query

### PR DESCRIPTION
**Pull request by OfTheDelmer**
Closes https://github.com/influxdata/platform/issues/1503

_Briefly describe your proposed changes:_
Add query name and editing field
_What was the problem?_
A user should be able to specify a name that refers to a query
_What was the solution?_
Update the TimeMachineTab name to be editable using either double click or right click.

  - [ ] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [ ] Rebased/mergeable
  - [ ] Tests pass

### CODE REVIEW

File: ui/src/types/v2/dashboards.ts [hunk 1]
```diff
@@ -100,19 +101,23 @@ export interface ViewLinks {
...
   | TableView
   | GaugeView
-
-export type ViewProperties =
-  | QueryViewProperties
   | MarkdownView
   | EmptyView
   | LogViewerView
 
+export type QueryViewProperties = Extract<
```
Comment (OfTheDelmer) [inline ui/src/types/v2/dashboards.ts]: [Conditional Typing](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-8.html)

```
Extract<T, U> – Extract from T those types that are assignable to U
```

Selects only `ViewProperties` with queries.

Comment (ischolten) [reply ui/src/types/v2/dashboards.ts]: woaaa

**Comment by OfTheDelmer** [comment]
I can open a separate issues to discuss two things that came up:

- Add a var to disable/enable right click actions (so we don't gotta have right click inspect in dev mode)
- Development assertions instead of throwing errors in clockface components

File: ui/src/App.tsx [hunk 2]
```diff
@@ -10,6 +11,7 @@ interface Props {
 const App: SFC<Props> = ({children}) => (
   <div className="chronograf-root">
     <Notifications />
+    <RightClickLayer />
```
Comment (ischolten) [inline ui/src/App.tsx]: cool

**Pull request merged**